src/psyclone/dynamo0p3.py

```diff
@@ -639,18 +666,25 @@ def __init__(self, arg_type):
                     "It does not make sense for a field on a discontinuous "
                     "space ({0}) to have a 'gh_inc' access".
                     format(self._function_space1.lower()))
-            # TODO: extend for "gh_write"
+            if self._function_space1.lower() in \
+               VALID_ANY_DISCONTINUOUS_SPACE_NAMES and \
+               self._access_type == AccessType.INC:
+                raise ParseError(
+                    "In the Dynamo0.3 API a field on any_discontinuous_space "
+                    "cannot have 'gh_inc' access because it is treated "
+                    "as discontinuous")
+            # TODO (issue #138): extend for "gh_write"
```

Comment:
Since we now have `ANY_DISCONTINUOUS_SPACE` I don't see the point of disallowing `GH_WRITE` for continuous spaces and not include `ANY_SPACE` in this restriction for the user-supplied kernels that iterate over cells.  `ANY_SPACE` is essentially treated as continuous and hence requires colouring when iterating over cells.
Of course, there is a question of dealing with built-ins as they loop over DoFs and not over cells. In their case using `GH_WRITE` (and `GH_READWRITE` instead of `GH_INC`) with `ANY_SPACE` makes sense. I opened the issue #471 for that.

I updated descriptions of #138 and #471 to distinguish between looping over cells and DoFs and modified TODO for `ANY_SPACE` to point to issue #471 instead.